### DISCUSSION TITLE
Repository "Languages" showing incorrect information

### DISCUSSION THREAD

**Discussion by 64J0** [community/community#54854] [Repositories] [answered]
### Select Topic Area

Bug

### Body

Hello, hope you're good.

I'm here to report what seems to be a bug in my perspective.

During this weekend, I have worked in a POC project to show how to use F# language to create custom GitHub Actions. And indeed, I was able to make it. One can find the repository by this link: https://github.com/64J0/custom-fsharp-gh-action-and-dynamic-matrix.

I have placed the custom Action code inside the `.github/fsharp-custom-action/` path, but this F# code is not appearing in the repository languages:

![image](https://user-images.githubusercontent.com/50725287/236817253-290bfd9b-a3d6-46f5-a317-99a6c1d971e4.png)

Indeed, I have some HCL code outside the .github/ directory, but I'd like to see the F# being mentioned there as well. When comparing the files, I guess that the portion of the F# code is even larger than the code of HCL.

**Comment by airtower-luna** [comment, marked as answer]
GitHub uses [Linguist](https://github.com/github-linguist/linguist) to detect languages, see its [troubleshooting guide](https://github.com/github-linguist/linguist/blob/master/docs/troubleshooting.md).

I'm pretty sure what's going on here is simply that Linguist [excludes `.github` by default](https://github.com/github-linguist/linguist/blob/e1c3110a4429d79deba370ceefda372dad03b003/lib/linguist/vendor.yml#L387) as "vendored". You could use [overrides in `.gitattributes` to change that](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md), but it might be better to restructure your sample action repository to look like a "normal" action repository, with `action.yml`, `src/`, etc. in the root of the repository.

**Reply by 64J0** [reply]
The point is that this repository is not a "normal" action repository. This action is not very general, so I think there will not be many people interested in using it the way it is.

> If you're building an action that you don't plan to make available to others, you can store the action's files in any location in your repository. If you plan to combine action, workflow, and application code in a single repository, we recommend storing actions in the `.github` directory. For example, `.github/actions/action-a` and `.github/actions/action-b`. -- [link](https://docs.github.com/en/actions/creating-actions/about-custom-actions#choosing-a-location-for-your-action).

But thank you very much for the reply, I'll take a look at the links you pointed there!

**Reply by 64J0** [reply]
I decided to open an [issue](https://github.com/github-linguist/linguist/issues/6405) in the Linguist repository as well, thanks for pointing it @airtower-luna!

Also, I added a [.gitattributes](https://github.com/64J0/custom-fsharp-gh-action-and-dynamic-matrix/blob/master/.gitattributes) to the repository, but it did not work as I'd like.

**Reply by airtower-luna** [reply]
As I understand the [relevant section](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md#vendored-code) need to use `-linguist-vendored` to mark it as *not vendored* instead of `linguist-detectable`. Also use `**` instead of `*` to catch files within subdirectories:

```gitattributes
.github/fsharp-custom-action/** -linguist-vendored
```

I'd expect that issue to get closed as "not a bug", because the path you're using is intentionally ignored by default. :sweat_smile:

**Reply by 64J0** [reply]
Got it, thanks for the heads-up. I also found [this issue](https://github.com/github-linguist/linguist/issues/5117) telling that it takes some hours to recompute the languages in the repository.

**Reply by 64J0** [reply]
Indeed, it was fairly fast. The language is already appearing to me. Thanks for the help @airtower-luna.

![image](https://user-images.githubusercontent.com/50725287/236893770-9aa0176c-b902-4407-af8a-466e7b09d279.png)

**Reply by raufzer** [reply]
Thanks!
